src/main/java/org/folio/circulation/domain/policy/lostitem/LostItemPolicy.java

```diff
@@ -121,10 +120,9 @@ public ChargeableFee getActualCostFee() {
   }
 
   public boolean shouldRefundFees(DateTime lostDateTime) {
-    final Period overdueMinutesInterval = minutesBetweenDateAndNow(lostDateTime);
-
     return feeRefundInterval == null
-      || overdueMinutesInterval.isLessThanOrEqualTo(feeRefundInterval);
+      || feeRefundInterval.hasNotPassedSinceDateTillNow(lostDateTime)
+      || feeRefundInterval.isPassingNow(lostDateTime);
```

Comment:
Can you help me understand these statements, this reads to me as though the interval could be either not have passed or has passed, which seems odd to me?

Reply:
The interval should pass before now or exactly now. If the interval has passed (lostDateTime + period > now) then no refund is allowed.

These statements are equivalent to -> `now().isBeforeOrEqualTo(lostDateTime + period)`.

Reply:
If the amount of time between the due date and now is exactly the same as the duration defined by the period (at that moment) has the period passed?

Reply:
Yes, it is treated as passed. 

has passed means - lostDate + period **>=** now()
has NOT passed means - lostDate + period **<** now();

Reply:
> Yes, it is treated as passed.
> 
> has passed means - lostDate + period `>=` now()
> has NOT passed means - lostDate + period `<` now();

Then wouldn't the first method `hasNotPassed...` be sufficient for this check?

Reply:
Hmm... we should also allow refund when interval is passing now (i.e. lostDate + period == now()). That's why we need the `isPassingNow` check. 

As you can see, the `hasNotPassed...` only checks if the interval strictly before now(), if it on the current date time than it is passed and the `hasNotPassed...` method returns `false` -> refund is disallowed, but should be allowed.

Am I missing something?

Reply:
Maybe I'm not understanding this logic. And I just realised I may not have followed your previous comment

To me, a period having passed is `lostDate + period > now()` because that means that at least the amount of time in the period has occurred. Maybe that is a little too convenient.

If this distinction is important, then I wonder if it's clearer to express it using those comparisons e..g `feeRefundInteval.addTo(lostDateTime) <= Clock.now()`?

Alternatively, I think maybe the language in the method names needs to change, I found it hard to understand what `passingNow` meant.

Reply:
> Alternatively, I think maybe the language in the method names needs to change, I found it hard to understand what passingNow meant.

What would you suggest as an alternative name?

Reply:
> What would you suggest as an alternative name?

I don't know, maybe something like `isEqualToDateTillNow`

I think this challenge of language is maybe a symptom of overreaching the scope of the period class.

Reply:
Renamed